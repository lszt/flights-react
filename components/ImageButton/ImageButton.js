import React, { PropTypes, Component } from 'react';
import classNames from 'classnames';
import './ImageButton.scss';

class ImageButton extends Component {

  render() {
    return (
      <div className={classNames(this.props.className, 'ImageButton')}>
        <a className="img-link" href={this.props.href}><img src={this.props.img}/></a>
        <a className="text-link" href={this.props.href}>{this.props.label}</a>
      </div>
    );
  }
}

ImageButton.propTypes = {
  className: PropTypes.string,
  label: PropTypes.string,
  img: PropTypes.string,
  href: PropTypes.string,
};

export default ImageButton;
