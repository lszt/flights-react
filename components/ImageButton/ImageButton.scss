.ImageButton {
  text-align: center;
  font-size: 2em;

  .img-link {
    display: block;
  }

  .text-link {
    display: block;
  }
}
