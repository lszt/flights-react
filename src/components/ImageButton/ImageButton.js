import React, {PropTypes} from 'react';
import styled from 'styled-components';
import {Link} from 'react-router-dom';

const Wrapper = styled.div`
  text-align: center;
  font-size: 2em;
`;

const StyledLink = styled(Link)`
  display: block;
  cursor: pointer;
`;

class ImageButton extends React.PureComponent {

  render() {
    const props = this.props;
    return (
      <Wrapper className={props.className}>
        <StyledLink to={props.href} onClick={props.onClick}><img src={props.img}/></StyledLink>
        <StyledLink to={props.href} onClick={props.onClick}>{props.label}</StyledLink>
      </Wrapper>
    )
  }
}

ImageButton.propTypes = {
  className: PropTypes.string,
  label: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  href: PropTypes.string,
  onClick: PropTypes.func,
};

export default ImageButton;
