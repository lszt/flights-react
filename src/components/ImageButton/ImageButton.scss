.ImageButton {
  text-align: center;
  font-size: 2em;

  .img-link {
    display: block;
    cursor: pointer;
  }

  .text-link {
    display: block;
    cursor: pointer;
  }
}
