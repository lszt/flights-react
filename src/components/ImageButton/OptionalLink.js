import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import {Link} from 'react-router-dom';

const StyledLink = styled(Link)`
  display: block;
  cursor: pointer;
`;

const ClickableChild = styled.div`
  cursor: pointer;
`;

const OptionalLink = props =>
  props.href
    ? <StyledLink to={props.href} onClick={props.onClick}>{props.children}</StyledLink>
    : <ClickableChild onClick={props.onClick}>{props.children}</ClickableChild>;

OptionalLink.propTypes = {
  href: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node,
};

export default OptionalLink;
