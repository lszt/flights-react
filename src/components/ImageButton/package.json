{
  "name": "ImageButton",
  "version": "0.0.0",
  "private": true,
  "main": "./ImageButton.js"
}
