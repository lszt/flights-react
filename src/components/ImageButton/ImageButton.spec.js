import React from 'react';
import TestUtils from 'react-dom/test-utils';
import expect from 'expect';
import ImageButton from './ImageButton';
import {MemoryRouter, Link} from 'react-router-dom';

describe('components', () => {
  describe('ImageButton', () => {
    it('renders Link components if `href` prop is set', () => {
      const testComponent = (
        <MemoryRouter>
          <ImageButton href="/some/path"/>
        </MemoryRouter>
      );
      const wrapper = TestUtils.renderIntoDocument(testComponent);

      const links = TestUtils.scryRenderedComponentsWithType(wrapper, Link);
      expect(links.length).toBe(2);
    });

    it('renders no Link components if `href` prop is not set', () => {
      const testComponent = (
        <MemoryRouter>
          <ImageButton onClick={() => {}}/>
        </MemoryRouter>
      );
      const wrapper = TestUtils.renderIntoDocument(testComponent);

      const links = TestUtils.scryRenderedComponentsWithType(wrapper, Link);
      expect(links.length).toBe(0);
    });
  });
});
