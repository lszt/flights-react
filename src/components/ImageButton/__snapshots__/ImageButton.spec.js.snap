// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ImageButton propagates href to OptionalLink 1`] = `
"<styled.div className={[undefined]}>
  <OptionalLink href=\\"/some/path\\" onClick={[undefined]} dataCy={[undefined]}>
    <img src=\\"\\" />
  </OptionalLink>
  <OptionalLink href=\\"/some/path\\" onClick={[undefined]} />
</styled.div>"
`;

exports[`components ImageButton propagates onClick to OptionalLink 1`] = `
"<styled.div className={[undefined]}>
  <OptionalLink href={[undefined]} onClick={[Function: onClick]} dataCy={[undefined]}>
    <img src=\\"\\" />
  </OptionalLink>
  <OptionalLink href={[undefined]} onClick={[Function: onClick]} />
</styled.div>"
`;

exports[`components ImageButton renders label and image 1`] = `
"<styled.div className={[undefined]}>
  <OptionalLink href={[undefined]} onClick={[undefined]} dataCy={[undefined]}>
    <img src=\\"imgsource.jpg\\" />
  </OptionalLink>
  <OptionalLink href={[undefined]} onClick={[undefined]}>
    my label
  </OptionalLink>
</styled.div>"
`;
